File: Day1/calorie_counting.rb
# frozen_string_literal: true

input = []
File.foreach('input.txt') { |line| input.push(line) }

input = input.join('').split(/\s{2,}/)
elfs = {}

input.each_with_index do |num, index|
  nums = num.split(' ').map(&:to_i)
  elfs[index] = nums.reduce(:+)
end

p elfs.values.max
p elfs.values.max(3).reduce(:+)

File: Day1/calorie_counting.py
input = open("input.txt", "r")

elves = []
text = []

for x in input:
    text.append(x)

text = "".join(text)
text = text.split("\n\n")

for id, line in enumerate(text):
    nums = line.split('\n')
    total = 0
    for num in nums:
        total += int(num,base=10)
    elves.append(total)

elves.sort(reverse=True)
print(elves[0])
print(elves[0] + elves[1] + elves[2])
